src/NzbDrone.Core/Datastore/Migration/198_parse_titles_from_existing_subtitle_files.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections.Generic;
+using System.Data;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Dapper;
+using FluentMigrator;
+using NzbDrone.Core.Datastore.Migration.Framework;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.MediaFiles.EpisodeImport.Aggregation.Aggregators;
+
+namespace NzbDrone.Core.Datastore.Migration
+{
+    [Migration(198)]
+    public class parse_title_from_existing_subtitle_files : NzbDroneMigrationBase
+    {
+        protected override void MainDbUpgrade()
+        {
+            Alter.Table("SubtitleFiles").AddColumn("Title").AsString().Nullable();
+            Alter.Table("SubtitleFiles").AddColumn("Copy").AsInt32().WithDefaultValue(0);
+            Execute.WithConnection(UpdateTitles);
+        }
+
+        private void UpdateTitles(IDbConnection conn, IDbTransaction tran)
+        {
+            var updates = new List<object>();
+
+            using (var cmd = conn.CreateCommand())
+            {
+                cmd.Transaction = tran;
+                cmd.CommandText = "SELECT \"Id\", \"RelativePath\", \"EpisodeFileId\", \"Language\", \"LanguageTags\" FROM \"SubtitleFiles\"";
+
+                using var reader = cmd.ExecuteReader();
+                while (reader.Read())
+                {
+                    var id = reader.GetInt32(0);
+                    var relativePath = reader.GetString(1);
+                    var episodeFileId = reader.GetInt32(2);
+
+                    var episodeFile = conn.QuerySingle<EpisodeFile>("SELECT * FROM \"EpisodeFiles\" WHERE \"Id\" = @Id", new { Id = episodeFileId });
+
+                    var subtitleTitleInfo = AggregateSubtitleInfo.CleanSubtitleTitleInfo(episodeFile, relativePath);
```

Comment:
The migration should be a snapshot in time and instead of calling this function we should extract the necessary pieces into a function in this migration, that way if the structure of episode file changes or we refactor `AggregateSubtitleInfo.CleanSubtitleTitleInfo` we don't break this migration.

Reply:
Well I changed that, however it does still depend on the `LanguageParser`. If I wanted to avoid that I would have to copy in several methods from that as well, which admittedly seems like a horrific amount of code duplication.

Reply:
Yeah, we do use parsers elsewhere, but their inputs are usually pretty basic, so should be safe, the extra copied code here does clean up the need for a full episode file at least.